Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs

```diff
@@ -51,33 +51,34 @@ public override void Initialize(AnalysisContext context)
 
 			context.RegisterCompilationStartAction(compilationContext =>
 			{
+				AllowedSymbols allowedSymbols = new();
 				EditorConfigOptions options = InitializeEditorConfigOptions(compilationContext.Options, compilationContext.Compilation, out Diagnostic configurationError);
-				HashSet<ISymbol> allowed = new();
 				if (!options.IgnoreExceptionsFile)
 				{
-					allowed = InitializeAllowed(compilationContext.Options.AdditionalFiles, compilationContext.Compilation);
+					allowedSymbols = InitializeAllowed(compilationContext.Options.AdditionalFiles, compilationContext.Compilation);
 				}
-				var compilationAnalyzer = new CompilationAnalyzer(options.TokenCount, allowed, options.GenerateExceptionsFile, configurationError);
+				var compilationAnalyzer = new CompilationAnalyzer(options.TokenCount, allowedSymbols, options.GenerateExceptionsFile, configurationError);
 				compilationContext.RegisterSyntaxNodeAction(compilationAnalyzer.AnalyzeMethod, SyntaxKind.MethodDeclaration);
 				compilationContext.RegisterCompilationEndAction(compilationAnalyzer.EndCompilationAction);
 			});
 		}
 
 		public const string AllowedFileName = @"DuplicateCode.Allowed.txt";
 
-		public virtual HashSet<ISymbol> InitializeAllowed(ImmutableArray<AdditionalText> additionalFiles, Compilation compilation)
+		public virtual AllowedSymbols InitializeAllowed(ImmutableArray<AdditionalText> additionalFiles, Compilation compilation)
 		{
+			AllowedSymbols allowedSymbols = new();
 			foreach (AdditionalText additionalFile in additionalFiles)
 			{
 				string fileName = Path.GetFileName(additionalFile.Path);
 				StringComparer comparer = StringComparer.OrdinalIgnoreCase;

```

Comment:
This declaration can be outside the loop

Reply:
refactored